core/embed/rust/src/micropython/recovery_type.rs

```diff
@@ -0,0 +1,40 @@
+#[cfg(feature = "micropython")]
+use crate::{
+    error::{value_error, Error},
+    micropython::obj::Obj,
+};
+
+/// RecoveryType as defined in `common/protob/messages-management.proto`,
+/// used as arguments coming from micropython into rust world for layouts or
+/// flows.
+pub enum RecoveryType {
+    Normal = 0,
+    DryRun = 1,
+    UnlockRepeatedBackup = 2,
+}
+
+// Converting `Obj` into `RecoveryType` enum
+#[cfg(feature = "micropython")]
+impl TryFrom<Obj> for RecoveryType {
+    type Error = Error;
+
+    fn try_from(obj: Obj) -> Result<Self, Self::Error> {
+        let val = u32::try_from(obj)?;
+        let this = Self::try_from(val)?;
+        Ok(this)
+    }
+}
+
+// Converting `u32` to `RecoveryType`
+impl TryFrom<u32> for RecoveryType {
+    type Error = Error;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(RecoveryType::Normal),
+            1 => Ok(RecoveryType::DryRun),
+            2 => Ok(RecoveryType::UnlockRepeatedBackup),
+            _ => Err(value_error!(c"Invalid RecoveryType")),
+        }
+    }
+}
```

Comment:
Appreciate putting the enum in its own file but it doesn't belong under `crate::micropython`. Currently the best place is under `crate::ui::layout::util`.

Reply:
My thought was that this value is utilized only to convert parameters coming from uPy so why not putting it into `crate::micropython`. But I guess you're right that util is more suitable for now. Even though I can imagine moving it out of `crate::ui` in the future as it's not just UI related.
ba3332c2067f739acb727065d979da84f73ad8eb

Reply:
I think `crate::micropython` is currently one of the tidier sub-crates, compared to e.g. UI, let's keep it that way:D.

Conceptually the enum belongs somewhere under application-specific constants, which we currently don't have. But ultimately it should be generated from protobuf along with e.g. ButtonRequestType.